### PR TITLE
chore: bump Go 1.22 and run `go mod tidy`

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
## Why

First of all, thank you for your amazing job.
This provider (module) makes it easier to handle the ACME challenges.

To maintain to the latest version as much as possible, I ran `go mod tidy -go=1.22`.

Thank you in advance.

### CODE REVIEW

**Comment by KeisukeYamashita** [comment]
@aymanbagabas Hello, sorry for bothering you. 
Could you please take a look when you get a chance?

Thank you for your wonderful project!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
   build:
     strategy:
       matrix:
-        go-version: [^1, ~1.19]
+        go-version: [^1, ~1.22]
```
Comment (aymanbagabas) [inline .github/workflows/build.yml:16]:
```suggestion
        go-version: [^1, ~1.21]
```

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/caddy-dns/route53
 
-go 1.19
+go 1.22
```
Comment (aymanbagabas) [inline go.mod:3]:
```suggestion
go 1.21
```

**Review by aymanbagabas — APPROVED** [review]
Match caddyserver go version

**Pull request merged**